execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -315,6 +315,73 @@ where
     Ok(missing_descendants)
 }
 
+/// Given a serialized trie, find any children that are referenced but not present in the database.
+pub fn missing_children<K, V, T, S, E>(
+    _correlation_id: CorrelationId,
+    txn: &T,
+    store: &S,
+    trie_raw: &[u8],
+) -> Result<Vec<Digest>, E>
+where
+    K: ToBytes + FromBytes + Eq + std::fmt::Debug,
+    V: ToBytes + FromBytes + std::fmt::Debug,
+    T: Readable<Handle = S::Handle>,
+    S: TrieStore<K, V>,
+    S::Error: From<T::Error>,
+    E: From<S::Error> + From<bytesrepr::Error>,
+{
+    // Optimization: Don't deserialize leaves as they have no descendants.
+    if let Some(&Trie::<K, V>::LEAF_TAG) = trie_raw.first() {
+        return Ok(vec![]);
+    }
+
+    // Parse the trie, handling errors gracefully.
+    let trie = match bytesrepr::deserialize_from_slice(trie_raw) {
+        Ok(trie) => trie,
+        // Couldn't parse; treat as missing and continue.
+        Err(err) => {
+            error!(?err, "unable to parse trie");
+            return Err(err.into());
+        }
+    };
+
+    let is_present = |trie_key| matches!(store.get_raw(txn, &trie_key), Ok(Some(_)));
```

Comment:
Is there a way to check for an entry's presence without loading all the data?

Reply:
I don't know - the code for finding missing descendants did it the same way, so I just assumed there isn't one.

Reply:
(It feels like we should consider adding one at some point; not in this PR, of course!)